File: CodeForces/minimize_the_error.py
'''
Examples
 input
2 0 0
1 2
2 3
 input
2
 input
2 1 0
1 2
2 2
 input
0
 input
2 5 7
3 4
14 4
 input
1
'''

n,k1,k2 = [int(x) for x in input().split()]
a = [int(x) for x in input().split()]
b = [int(x) for x in input().split()]
diffs = [abs(y-x) for x, y in zip(a, b)]
for _ in range(k1+k2+1):
    print(diffs)
    diffs[0] = abs(sorted(diffs,reverse=True)[0] - 1)


print(sum([a*a for a in diffs]))

File: CodeForces/minimize_the_error.kt
fun main(args: Array<String>) {
    val (n ,k1, k2) = readLine()!!.split(' ').map { it.toInt() }
    val a = readLine()!!.split(' ').map { it.toLong() }
    val b = readLine()!!.split(' ').map { it.toLong() }

    val d = List(a.size, {Math.abs(b[it] - a[it])}).toMutableList()

    (1 .. k1 + k2).forEach {
        d.sortDescending()
        d[0] = Math.abs(d[0] - 1)
        println(d)
    }


    println(d.map { it * it }.sum())
}

fun List<Long>.sortMax():Int {
    for (i in 1 until this.size){
        if (this[i] < this[i-1]){
            return i-1
        }
    }
    return this.size-1
}